plugin/trino-mariadb/src/main/java/io/trino/plugin/mariadb/MariaDbClient.java

```diff
@@ -623,6 +647,102 @@ protected boolean isSupportedJoinCondition(ConnectorSession session, JdbcJoinCon
                 .noneMatch(type -> type instanceof CharType || type instanceof VarcharType);
     }
 
+    @Override
+    public TableStatistics getTableStatistics(ConnectorSession session, JdbcTableHandle handle)
+    {
+        if (!statisticsEnabled) {
+            return TableStatistics.empty();
+        }
+        if (!handle.isNamedRelation()) {
+            return TableStatistics.empty();
+        }
```

Comment:
`SHOW STATS FOR` goes to this path or does it happen inside engine?

Reply:
depends on "precalculate stats". for Maria connector, precalculate is true, so it shouldn't hit here